src/lib.rs

```diff
@@ -406,6 +414,44 @@ impl Watch {
                 .any(|x| x.to_string_lossy().ends_with('~'))
         })
     }
+
+    fn include_commands(&self, list: impl Into<CommandList>) -> CommandList {
+        let mut list = list.into();
+
+        if !self.shell_commands.is_empty() {
+            list.append(
+                &mut self
+                    .shell_commands
+                    .iter()
+                    .map(|x| {
+                        let mut command = Command::new("/bin/sh");
+                        command.arg("-c");
+                        command.arg(x);
+
+                        command
+                    })
+                    .collect(),
+            );
+        }
+
+        if !self.cargo_commands.is_empty() {
+            list.append(
+                &mut self
+                    .cargo_commands
+                    .iter()
+                    .map(|x| {
+                        let mut command = Command::new("/bin/sh");
+                        command.arg("-c");
+                        command.arg(format!("cargo {x}"));
+
+                        command
+                    })
+                    .collect(),
+            )
```

Comment:
You're collecting stuff into a Vec, then pass it as mutable ref to the list... OR you could use extend() instead of append and pass the iterator without collecting anything and thus avoiding useless allocation and code complexity.